src/extensions/default/AutoUpdate/main.js

```diff
@@ -0,0 +1,630 @@
+                postMessageToNode(MessageIds.INITIALIZE_STATE, _updateParams);
+            })
+            .fail(function () {
+                UpdateInfoBar.showUpdateBar({
+                    type: "error",
+                    title: Strings.INITIALISATION_FAILED,
+                    description: ""
+                });
+            });
+    }
+
+    /**
+     * Handles the completion of node state initialization
+     */
+    function handleInitializationComplete() {
+        enableCheckForUpdateEntry(false);
+        getLatestInstaller();
+    }
+
+    /**
+     * Handles the Cancel button click by user in
+     * Unsaved changes prompt, which would come up if user
+     * has dirty files and he/she clicked UpdateNow
+     */
+    function dirtyFileSaveCancelled() {
+        UpdateInfoBar.showUpdateBar({
+            type: "warning",
+            title: Strings.WARNING_TYPE,
+            description: Strings.UPDATE_ON_NEXT_LAUNCH
+        });
+    }
+
+    /**
+     * Handles the scenario where auto update in progress check returns error
+     */
+    function handleAutoUpdateError() {
+        resetStateInFailure("AutoUpdate : Error returned in auto update in progress check");
+    }
+
+    /**
+     * Handles Download completion callback from Node
+     */
+    function handleDownloadSuccess() {
+        UpdateStatus.showUpdateStatus("validating-installer");
+        validateChecksum();
+    }
+
+    /**
+     * Receives messages from node
+     * @param {object} event  - event received from node
+     * @param {object}   msgObj - json containing - {
+     *                          fn - function to execute on Brackets side
+     *                          args - arguments to the above function
+     */
+    function receiveMessageFromNode(event, msgObj) {
+        functionMap[msgObj.fn].apply(null, msgObj.args);
+    }
+
+    /**
+     * Generates a map for brackets side functions
+     */
+    function registerBracketsFunctions() {
+        functionMap["brackets.notifyinitializationComplete"] = handleInitializationComplete;
+        functionMap["brackets.showStatusInfo"]               = showStatusInfo;
+        functionMap["brackets.notifyDownloadSuccess"]        = handleDownloadSuccess;
+        functionMap["brackets.showErrorMessage"]             = showErrorMessage;
+        functionMap["brackets.notifyDownloadFailure"]        = handleDownloadFailure;
+        functionMap["brackets.notifySafeToDownload"]         = handleSafeToDownload;
+        functionMap["brackets.notifyvalidationStatus"]       = handleValidationStatus;
+    }
+
+    functionMap["brackets.registerBracketsFunctions"] = registerBracketsFunctions;
+
+    UpdateNotification.on(UpdateNotification.GET_AUTOUPDATE_INSTALLER, handleGetNow);
+    UpdateNotification.on(UpdateNotification.DIRTY_FILESAVE_CANCELLED, dirtyFileSaveCancelled);
+    UpdateNotification.on(UpdateNotification.AUTOUPDATE_ERROR, handleAutoUpdateError);
+
+    if(brackets.platform !== "linux") {
```

Comment:
How about checking for the shell functions existence like `setUpdateParamsAndQuit`, and if not present, just bail out?